CO-2511 Field getting days since last letter

Alaggaerion opened the pull request:
Added an invisible field to compute how many days passed since the last letter from the sponsor to the child
Prod templates still have to be updated in case the sponsor recently wrote to the child

Alaggaerion committed:
CO-2511
Added an invisible field to compute how many days passed since the last letter from the sponsor to the child

ecino commented on contract_compassion/views/contract_view.xml:
```diff
@@ -83,6 +83,7 @@
                         <field name="group_freq"/>
                         <field name="months_paid"/>
                         <field name="last_paid_invoice_date"/>
+                        <field name="last_letter" invisible="1"/>
```
It's not needed to add it in the view if you keep it invisible. Either remove it or keep it visible.

ecino commented on contract_compassion/models/contracts.py:
```diff
@@ -89,6 +89,9 @@ class RecurringContract(models.Model):
                                        help='Contract duration in days')
 
+    # Field used to compute days number since last sponsor's letter
+    last_letter = fields.Integer(compute='_compute_last_letter')
```
add a `help` parameter describing what the field is instead of using a comment above.

ecino commented on contract_compassion/models/contracts.py:
```diff
@@ -215,6 +218,18 @@ def _compute_contract_duration(self):
+        compute since how many days he didn't write him any letter
+        :return: int value
+        """
+        self.last_letter = (date.today() - datetime.strptime(
```
The compute method should always be able to be invoked on a recordset. This means that you should not compute only for one record but put a for loop and setting the values on each record:
```
for contract in self:
    contract.last_letter = ....
```

ecino replied:
Instead of using `datetime.strptime`, use `fields.Datetime.from_string()` method

ecino commented on contract_compassion/models/contracts.py:
```diff
@@ -215,6 +218,18 @@ def _compute_contract_duration(self):
+        self.last_letter = (date.today() - datetime.strptime(
+            self.env['correspondence'].search([(
+                'direction', '=', 'Supporter To Beneficiary'), (
+                'sponsorship_id', '=', self.id)], limit=1).scanned_date,
```
Did you think of the case if there are no letters? It should not fail but return the number of days since the creation of the sponsorship (using the field create_date)

Alaggaerion committed:
CO-2511
Code corrections and added try/except to deal with sponsorships without any letters or create_date

ecino commented:
flake8 issues:
```
contract_compassion/models/contracts.py:235:25: E126 continuation line over-indented for hanging indent
contract_compassion/models/contracts.py:244:29: E126 continuation line over-indented for hanging indent
contract_compassion/models/contracts.py:250:5: E303 too many blank lines (2)
```

Alaggaerion committed:
CO-2511
Corrected flake8 issues
